### ISSUE TITLE
Can't make it to work Terratonic with Terralith on paper server

### ISSUE THREAD

**Issue by gnoby** [Apollounknowndev/tectonic#231]
I try to get Terratonic-Data-Pack-1.21.zip with Terralith_1.21_v2.5.5.zip to run on my newly setup paper-1.21.1-119.jar server.

What I do:

- Exexute paper-1.21.1-119.jar for the first time to extract things then edit the eula text.
- I delete the directory contents of world, world_nether and world_the_end.
- I create a directory world/datapacks in the minecraft directory
- I copy Terratonic-Data-Pack-1.21.zip and Terralith_1.21_v2.5.5.zip into it.
- I rename Terralith_1.21_v2.5.5.zip to Tarralith_1.21_v2.5.5.zip so it gets loaded first
- I edit the server.properties line initial-enabled-packs=file/Tarratonic-Data-Pack-1.21.zip,file/Terralith_1.21_v2.5.5.zip,vanilla if I don't do that I get an exeption at the startup of the server
- I start the server java -Xms1G -Xmx8G -jar paper-1.21.1-119.jar
- I check what datapacks are loaded in what order:  "There are 5 data pack(s) enabled: [file/Tarratonic-Data-Pack-1.21.zip (world)], [file/Terralith_1.21_v2.5.5.zip (world)], [vanilla (built-in)], [file/bukkit (world)], [paper (built-in)]"
- I try it wih my client and get vanilla world generation :-(

Note: if I don't edit the server.properties initial-enabled-packs I get this exception
`java.lang.StackOverflowError: null
        at net.minecraft.world.level.levelgen.RandomState$1NoiseWiringHelper.visitNoise(RandomState.java:82) ~[paper-1.21.1.jar:1.21.1-119-7cd4f2c]
        at net.minecraft.world.level.levelgen.DensityFunctions$ShiftA.mapAll(DensityFunctions.java:956) ~[paper-1.21.1.jar:1.21.1-119-7cd4f2c]
        at net.minecraft.world.level.levelgen.DensityFunctions$HolderHolder.mapAll(DensityFunctions.java:604) ~[paper-1.21.1.jar:1.21.1-119-7cd4f2c]`

Edit just tried it with vanilla fabric server and get the same effect

**Comment by Apollounknowndev** [comment]
No renaming needs to be done. Terralith needs to load first and will load first because `Terral...` comes before `Terrat...`.

**Comment by gnoby** [comment]
__@Apollounknowndev Thank you then this is stated false on the modrinth page. https://modrinth.com/datapack/terratonic
_To use this datapack, Terratonic must load above Terralith in the datapack list! Tectonic is not needed, as everything from Tectonic is in Terratonic._

**Comment by Apollounknowndev** [comment]
What was in the modrinth page was true but misleading in the context of servers. I have a better explanation of how to install Terratonic [here](https://github.com/Apollounknowndev/tectonic/wiki/Terratonic-FAQ#how-do-i-install-terratonic) which the Terratonic page now redirects to.

If you have both datapacks installed and the server is generating vanilla terrain, either you have another datapack that is overriding worldgen or it *is* generating correctly but you haven't seen enough of the terrain yet (mountain peaks are a good indicator)

**Comment by gnoby** [comment]
Thank you! So to test it at first I installed the two packs on the "client internal server" which works just by putting the 2 packs into the mods folder and use a specific seed, so I could see how the world is supposed to look like. With this seed for example 85425578410475571 you get a big snowy mountain right away, which would never be in vanilla world generation.

I fixed it and it works now! On every try I only cleared the world folder except word/datapacks/ and the 2 zips. But I noticed on startup that it said "Missing data pack" and not stated what data pack is missing. Don't know what went wrong here and how a datapack could be missing since I didn't specify any. So I deleted the paper folder and started from scratch. This time without renaming the Terratonic file. And that is all what was needed. Just put the 2 files in the right folder (and in my case edit the server.properties to include the seed). Now the big snowy mountain is created :-) Thanks for your help again!

![image](https://github.com/user-attachments/assets/63d992e0-d645-485f-ac18-b9df44cc67ca)
